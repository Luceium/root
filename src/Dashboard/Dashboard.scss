.dashboard {
    width: 100%;
    height: 100vh;
    background-repeat: no-repeat;

	.dashboard-design {
		color: white;
        text-align: center;
        background-color: #686e77;
        width: 100%;
        max-width: 500px;
        padding: 20px;
	}
}

/*.dashboard-body {
    background-image: url(/art/circuit_left.svg), url(/art/circuit_bottom.svg);
    background-position: left top, right bottom;
    background-repeat: no-repeat, no-repeat;
    height: 100%;
}*/