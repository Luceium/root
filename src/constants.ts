export var STATUS = {
  INCOMPLETE: "incomplete",
  SUBMITTED: "submitted",
  WAITLISTED: "waitlisted",
  REJECTED: "rejected",
  ADMITTED: "admitted",
  ADMISSION_CONFIRMED: "admission_confirmed",
  ADMISSION_DECLINED: "admission_declined"
}

export var TYPE = {
  IN_STATE: "is",
  OUT_OF_STATE: "oos",
  STANFORD: "stanford"
}

export var DEADLINES = [
  {
    "key": "oos",
    "label": "out-of-state",
    "date": "2018-11-20T07:59:00.000Z",
    "display_date": "November 19, 2018"
  },
  {
    "key": "is",
    "label": "in-state",
    "date": "2018-11-27T07:59:00.000Z",
    "display_date": "November 26, 2018"
  },
  {
    "key": "stanford",
    "label": "Stanford student",
    "date": "2019-02-14T07:59:00.000Z",
    "display_date": "February 13, 2019"
  }
];

export var TRANSPORTATION_STATUS = {
  PENDING: "pending",
  UNAVAILABLE: "unavailable",
  AVAILABLE: "available",
  SUBMITTED: "submitted",
  REJECTED: "rejected",
  APPROVED: "approved",
  PAID: "paid"
};

export var TRANSPORTATION_TYPES = {
  BUS: "bus",
  FLIGHT: "flight",
  OTHER: "other"
};

export var TRANSPORTATION_BUS_ROUTES = {
  TEST: "test",
  USC: "usc",
  UCLA: "ucla",
  SANDIEGO: "sandiego",
  UCI: "uci",
  POMONA: "pomona",
  BERKELEY: "berkeley"
};

export var TRANSPORTATION_DEADLINES = {
  [TRANSPORTATION_TYPES.FLIGHT]: 'December 9th at 11:59pm PST',
  [TRANSPORTATION_TYPES.BUS]: 'January 10th at 11:59pm PST',
  [TRANSPORTATION_TYPES.OTHER]: 'February 19th at 11:59pm PST'
};


export type IBusRoute = {
  day?: string,
  time?: string,
  stop?: string,
  location?: string,
  hack?: boolean
};

export var TRANSPORTATION_BUS_ROUTE_DETAILS: { [x: string]: IBusRoute[] } = {
  [TRANSPORTATION_BUS_ROUTES.USC]: [
    {
      day: 'Friday, February 15th',
      time: '8:00am',
      stop: 'Check in at 37th & McClintock',
      location: 'Los Angeles'
    },
    {
      day: 'Friday, February 15th',
      time: '4:30pm',
      stop: 'Panama St. & Via Ortega',
      location: 'Stanford'
    },
    {
      hack: true
    },
    {
      day: 'Sunday, February 17th',
      time: '4:00pm',
      stop: 'Check in at Panama St. & Via Ortega',
      location: 'Stanford'
    },
    {
      day: 'Sunday, February 17th',
      time: '11:00pm',
      stop: '37th & McClintock',
      location: 'Los Angeles'
    }
  ],
  [TRANSPORTATION_BUS_ROUTES.TEST]: [
    {
      day: 'Friday, February 15th',
      time: '8:00am',
      stop: 'Check in at 37th & McClintock',
      location: 'Atlanta'
    },
    {
      hack: true
    },
    {
      day: 'Sunday, February 17th',
      time: '11:00pm',
      stop: '37th & McClintock',
      location: 'Los Angeles'
    }
  ],
  [TRANSPORTATION_BUS_ROUTES.TEST]: [
    {
      day: 'Friday, February 15th',
      time: '9:00am',
      stop: 'Check in at 10 Charles E. Young',
      location: 'Los Angeles'
    },

    {
      day: 'Friday, February 15th',
      time: '4:30pm',
      stop: 'Panama St. & Via Ortega',
      location: 'Stanford'
    },

    {
      hack: true
    },

    {
      day: 'Sunday, February 17th',
      time: '4:00pm',
      stop: 'Check in at Panama St. & Via Ortega',
      location: 'Stanford'
    },

    {
      day: 'Sunday, February 17th',
      time: '10:30pm',
      stop: '10 Charles E. Young',
      location: 'Los Angeles'
    }
  ],
  [TRANSPORTATION_BUS_ROUTES.SANDIEGO]: [
    {
      day: 'Friday, February 15th',
      time: '7:30am',
      stop: 'Check in at UCSD (location added later)',
      location: 'Los Angeles'
    },

    {
      day: 'Friday, February 15th',
      time: '6:00pm',
      stop: 'Panama St. & Via Ortega',
      location: 'Stanford'
    },

    {
      hack: true
    },

    {
      day: 'Sunday, February 17th',
      time: '3:30pm',
      stop: 'Check in at Panama St. & Via Ortega',
      location: 'Stanford'
    },

    {
      day: 'Sunday, February 17th',
      time: '11:45pm',
      stop: 'UCSD',
      location: 'Los Angeles'
    }
  ],
  [TRANSPORTATION_BUS_ROUTES.UCI]: [
    {
      day: 'Friday, Februrary 15th',
      time: '8:45am',
      stop: 'Check in at UCI (location added later)',
      location: 'Irvine'
    },

    {
      day: 'Friday, Feburary 15th',
      time: '6:00pm',
      stop: 'Panama St. & Via Ortega',
      location: 'Stanford'
    },

    {
      hack: true
    },

    {
      day: 'Sunday, Februrary 17th',
      time: '3:30pm',
      stop: 'Check in at Panama St. & Via Ortega',
      location: 'Stanford'
    },

    {
      day: 'Sunday, Februrary 17th',
      time: '11:00pm',
      stop: 'UCI',
      location: 'Irvine'
    }
  ],
  [TRANSPORTATION_BUS_ROUTES.POMONA]: [
    {
      day: 'Friday, Februrary 15th',
      time: '10:45am',
      stop: 'Check in at Harvey Mudd (location added later)',
      location: 'Claremont'
    },

    {
      day: 'Friday, Feburary 15th',
      time: '6:00pm',
      stop: 'Panama St. & Via Ortega',
      location: 'Stanford'
    },

    {
      hack: true
    },

    {
      day: 'Sunday, Februrary 17th',
      time: '3:30pm',
      stop: 'Check in at Panama St. & Via Ortega',
      location: 'Stanford'
    },

    {
      day: 'Sunday, Februrary 17th',
      time: '10:00pm',
      stop: 'Harvey Mudd',
      location: 'Claremont'
    }
  ],
  [TRANSPORTATION_BUS_ROUTES.BERKELEY]: [
    {
      day: 'Friday, Februrary 15th',
      time: '2:30pm',
      stop: 'Check in at West Circle / University Drive',
      location: 'Berkeley'
    },

    {
      day: 'Friday, Feburary 15th',
      time: '4:30pm',
      stop: 'Panama St. & Via Ortega',
      location: 'Stanford'
    },

    {
      hack: true
    },

    {
      day: 'Sunday, Februrary 17th',
      time: '3:30pm',
      stop: 'Check in at Panama St. & Via Ortega',
      location: 'Stanford'
    },

    {
      day: 'Sunday, Februrary 17th',
      time: '10:00pm',
      stop: 'West Circle / University Drive',
      location: 'Berkeley'
    }
  ]
};

export const HACKATHON_YEAR = 2019;
export const HACKATHON_DATE_RANGE = "February 15-17";