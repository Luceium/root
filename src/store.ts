import { Reducer, createStore, applyMiddleware } from 'redux';
import { reducers } from './store/index';
import thunkMiddleware from 'redux-thunk'
import { connectRouter, routerMiddleware, LOCATION_CHANGE } from 'connected-react-router';
import GoogleAnalyticsGtag, { trackPageView, trackEvent } from '@redux-beacon/google-analytics-gtag';
import { createMiddleware } from "redux-beacon";
import history from "./history";

declare const GA_TRACKING_ID: string;
const ga = GoogleAnalyticsGtag(GA_TRACKING_ID);
const eventsMap = {
  [LOCATION_CHANGE]: trackPageView(action => ({
    path: action.payload.routerState.url
  })),
  '*': trackEvent(action => ({
    category: 'redux',
    action: action.type,
  })),
}
const gaMiddleware = createMiddleware(eventsMap, ga);

const store = createStore(
  connectRouter(history)(reducers),
  applyMiddleware(thunkMiddleware, routerMiddleware(history), gaMiddleware)
);
export default store;