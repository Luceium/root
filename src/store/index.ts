import { combineReducers, Reducer } from 'redux';

import authReducer from './auth/reducer';
import baseReducer from './base/reducer';

export const reducers: Reducer = combineReducers({
  auth: authReducer,
  base: baseReducer
});
