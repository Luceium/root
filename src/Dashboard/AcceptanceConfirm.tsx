import React from "react";
import { HACKATHON_YEAR } from "../constants";
export default () => <div>
    <h4>Congratulations! You have been selected to attend Treehacks {HACKATHON_YEAR}!</h4>
    <p>You have until February 1st to confirm your attendance at Treehacks {HACKATHON_YEAR}. If you don't confirm by then, we'll assume you can't make it and give your spot to someone else.</p>
    <button className="btn btn-custom inverted">decline spot</button>
    <button className="btn btn-custom">confirm spot</button>
    <p>We are committed to helping every admitted hacker get here! Check out the travel section for details.</p>
</div>;