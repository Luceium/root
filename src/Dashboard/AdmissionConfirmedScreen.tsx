import React from "react";
import { HACKATHON_YEAR } from "../constants";

export default () => <div>
    <p>Admission confirmed! See you there.</p>
</div>;