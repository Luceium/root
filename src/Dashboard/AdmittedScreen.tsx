import React from "react";
import { connect } from "react-redux";
import { HACKATHON_YEAR } from "../constants";
import { confirmAdmission, declineAdmission } from "../store/form/actions";
import { Link } from "react-router-dom";
const moment = require("moment-timezone");

interface IAdmittedScreenProps {
    confirmAdmission: () => void,
    declineAdmission: () => void,
    confirmedYet: boolean,
    deadline: string
}
export const AdmittedScreen = (props: IAdmittedScreenProps) => <div className="admitted-content">
    {props.confirmedYet && <div>
        <h4>You're coming to Treehacks {HACKATHON_YEAR}!</h4>
        <p>Admission confirmed &ndash; see you there! Make sure to review and confirm your <Link to="/transportation">travel information</Link>.</p>
    </div>}
    {!props.confirmedYet && <div>
        <h4>Congratulations! You have been selected to attend Treehacks {HACKATHON_YEAR}!</h4>
        <p>You have until {moment(props.deadline).tz("America/Los_Angeles").format("LLL z")} to confirm your attendance at Treehacks {HACKATHON_YEAR}. If you don't confirm by then, we'll assume you can't make it and give your spot to someone else.</p>
    </div>}
    <button className="btn btn-custom inverted" onClick={() => window.confirm(`Are you sure you would like to decline your spot at Treehacks ${HACKATHON_YEAR}? Your decision is final.`) && props.declineAdmission()}>decline spot</button>
    <button className="btn btn-custom" disabled={props.confirmedYet} onClick={() => props.confirmAdmission()}>{props.confirmedYet ? "already confirmed": "confirm spot"}</button>
    {!props.confirmedYet &&
        <p>We are committed to helping every admitted hacker get here! Check out the <Link to="/transportation">travel section</Link> for details.</p>
    }
</div>;

const mapStateToProps = state => ({

});

const mapDispatchToProps = (dispatch, ownProps) => ({
    confirmAdmission: () => dispatch(confirmAdmission()),
    declineAdmission: () => dispatch(declineAdmission())
});

export default connect(mapStateToProps, mapDispatchToProps)(AdmittedScreen);