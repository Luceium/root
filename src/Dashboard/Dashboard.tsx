import React from "react";
import { connect } from "react-redux";
import { getUserProfile } from "../store/form/actions";
import { IFormState } from "../store/form/types";
import { IDashboardProps, IDashboardWrapperProps } from "./types";
import Loading from "../Loading/Loading";
import "./Dashboard.scss";

function getDeadline(type) {
    switch (type) {
        case "is":
            return new Date("2018-11-27T07:59:00.000Z");
        case "stanford":
            return new Date("2019-02-14T07:59:00.000Z");
        case "oos":
        default:
            return new Date("2018-11-20T07:59:00.000Z");
    }
  }

function formatDate(date) {
    var monthNames = [
        "January", "February", "March",
        "April", "May", "June", "July",
        "August", "September", "October",
        "November", "December"
    ];

    var day = date.getDate();
    var monthIndex = date.getMonth();
    var year = date.getFullYear();

    return monthNames[monthIndex] + ' ' + day + ', ' + year;
}


export const Dashboard = (props: IDashboardProps) => {
    let date = getDeadline(props.profile.type);
    const dateNow = new Date();
    const diffDays = Math.round(Math.abs((date.getTime() - dateNow.getTime()) / (24 * 60 * 60 * 1000)));
    const deadline = formatDate(date);
    return (
        <div className="dashboard" style={{ "backgroundImage": `url('${require('../art/combined_circuit.svg')}')` }}>
            <div style={{ position: 'absolute', top: "50%", left: "50%", transform: "translateX(-50%) translateY(-50%)" }}>
                <div className="dashboard-design">
                    {
                        props.profile.status === "submitted" ? (
                            <span>
                                Your application has been received &ndash; you are all good for now!<br /><br />We will email you when decisions are released and will handle any travel questions at that time. Thanks for applying :)</span>
                        ) : (
                                <div>
                                    <div>
                                        You haven't submitted your application yet. You have
                            </div>
                                    <div style={{ color: "#00E073", fontSize: '70px', marginBottom: -15 }}>
                                        {diffDays}
                                    </div>
                                    <div>
                                        days to submit your application before the deadline:<br /><strong>{deadline}</strong>.
                            </div>
                                </div>
                            )
                    }
                </div>
            </div>
        </div>
    );
}

const mapStateToProps = state => ({
    ...state.form
});

const mapDispatchToProps = (dispatch, ownProps) => ({
    getUserProfile: () => dispatch(getUserProfile())
});

class DashboardWrapper extends React.Component<IDashboardWrapperProps, {}> {
    componentDidMount() {
        this.props.getUserProfile();
    }
    render() {
        if (!this.props.profile) {
            return <Loading />;
        }
        return <Dashboard profile={this.props.profile} />;
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(DashboardWrapper);
