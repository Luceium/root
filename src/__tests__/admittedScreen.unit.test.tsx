import React from "react";
import { shallow, mount, render } from 'enzyme';
import { Provider } from "react-redux";
import store from "../store";
import sinon from "sinon";
import lolex from "lolex";
import {Dashboard} from "../Dashboard/Dashboard";
import { STATUS, TYPE, DEADLINES } from "../constants";
import {AdmittedScreen} from "../Dashboard/AdmittedScreen";

const defaultProps = {
    confirmAdmission: () => null,
    declineAdmission: () => null,
    confirmedYet: true,
    deadline: "2048-11-28T04:39:47.512Z"
}

it('Format admission deadline with PST time zone', () => {
    const wrapper = shallow(
        <AdmittedScreen {...defaultProps} confirmedYet={false} />
    );
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.text()).toContain("You have until November 27, 2048 8:39 PM PST to confirm your attendance");
    expect(wrapper.text()).toContain("Congratulations! You've been accepted to TreeHacks 2019!");
});

it('Admission confirmed screen', () => {
    const wrapper = shallow(
        <AdmittedScreen {...defaultProps} confirmedYet={true} />
    );
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.text()).toContain("Admission confirmed â€“ see you there!");
    expect(wrapper.text()).toContain("decline spot");
    expect(wrapper.text()).toContain("already confirmed");
});