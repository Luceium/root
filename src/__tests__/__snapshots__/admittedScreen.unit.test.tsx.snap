// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Admission confirmed screen 1`] = `
<div
  className="admitted-content"
>
  <div>
    <h4>
      You're coming to Treehacks 
      2019
      !
    </h4>
    <p>
      Admission confirmed â€“ see you there! Make sure to review and confirm your 
      <Link
        replace={false}
        to="/transportation"
      >
        travel information
      </Link>
      .
    </p>
  </div>
  <button
    className="btn btn-custom inverted"
    onClick={[Function]}
  >
    decline spot
  </button>
  <button
    className="btn btn-custom"
    disabled={true}
    onClick={[Function]}
  >
    already confirmed
  </button>
</div>
`;

exports[`Format admission deadline with PST time zone 1`] = `
<div
  className="admitted-content"
>
  <div>
    <h4>
      Congratulations! You have been selected to attend Treehacks 
      2019
      !
    </h4>
    <p>
      You have until 
      November 27, 2048 8:39 PM PST
       to confirm your attendance at Treehacks 
      2019
      . If you don't confirm by then, we'll assume you can't make it and give your spot to someone else.
    </p>
  </div>
  <button
    className="btn btn-custom inverted"
    onClick={[Function]}
  >
    decline spot
  </button>
  <button
    className="btn btn-custom"
    disabled={false}
    onClick={[Function]}
  >
    confirm spot
  </button>
  <p>
    We are committed to helping every admitted hacker get here! Check out the 
    <Link
      replace={false}
      to="/transportation"
    >
      travel section
    </Link>
     for details.
  </p>
</div>
`;
